docs/reference/memory-alloc.rst

```diff
@@ -8,23 +8,15 @@
 ********************************************************************
 Device memory allocation in rocBLAS
 ********************************************************************
+rocBLAS uses per-handle device memory allocation to manage temporary memory efficiently. Each handle maintains its own memory and executes kernels sequentially in a single stream, allowing memory reuse across kernels.
 
-For temporary device memory, rocBLAS uses a per-handle memory allocation with out-of-band management.
-The temporary device memory is stored in the handle. This allows for the recycling of temporary device memory
-across multiple computational kernels that use the same handle. Each handle has a single stream, and
-kernels execute in order in the stream, with each kernel completing before the next kernel in the
-stream starts. There are four schemes for temporary device memory:
+There are two memory allocation schemes:
 
-*  **rocBLAS_managed**: This is the default scheme. If there is not enough memory in the handle, computational functions allocate the required
-   memory. Any memory that is allocated persists in the handle, so it is available for later computational functions that use the handle.
-*  **user_managed, preallocate**: An environment variable is set before the rocBLAS handle is created.
-   After that there are no more allocations or deallocations.
-*  **user_managed, manual**:  The user calls helper functions to get or set the memory size throughout the program,
-   thereby controlling when allocation and deallocation occur.
-*  **user_owned**:  The user allocates workspace and calls a helper function to allow rocBLAS to access the workspace.
+*  **rocBLAS_managed(default)**: rocBLAS internally manages memory, allocating more if needed. Allocated memory persists with the handle for reuse.
 
-The default scheme has the disadvantage that allocation is synchronizing, so if there is not enough memory in the handle,
-a synchronizing deallocation and allocation occur.
+*  **user_owned**: Users allocate memory and provide it to rocBLAS via helper functions.
+
+Stream-ordered device memory allocation is the default. rocBLAS uses hipMallocAsync and hipFreeAsync to allocate and free memory in stream order, avoiding global synchronization. This enables seamless stream switching without needing hipStreamSynchronize().
```

Comment:
Should this be changed from:
Stream-ordered device memory allocation is the default. rocBLAS uses
to
rocBLAS_managed is the default. It uses